### ISSUE TITLE
[Bug]: message['content'] key error

### ISSUE THREAD

**Issue by zhuantouer** [All-Hands-AI/OpenHands#5142]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

the provider config
<img width="353" alt="image" src="https://github.com/user-attachments/assets/3e4b325a-75b6-4dbe-a1f9-42ed13e7300b">

### OpenHands Installation

Docker command in README

### OpenHands Version

0.14

### Operating System

Linux

### Logs, Errors, Screenshots, and Additional Context

```bash
Traceback (most recent call last):
  File "/app/openhands/controller/agent_controller.py", line 203, in start_step_loop
    await self._step()
  File "/app/openhands/controller/agent_controller.py", line 484, in _step
    action = self.agent.step(self.state)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/openhands/agenthub/codeact_agent/codeact_agent.py", line 334, in step
    response = self.llm.completion(**params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 336, in wrapped_f
    return copy(f, *args, **kw)
           ^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 475, in __call__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 376, in iter
    result = action(retry_state)
             ^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 398, in <lambda>
    self._add_action_func(lambda rs: rs.outcome.result())
                                     ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 478, in __call__
    result = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "/app/openhands/llm/llm.py", line 196, in wrapper
    messages = convert_fncall_messages_to_non_fncall_messages(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/openhands/llm/fn_call_converter.py", line 323, in convert_fncall_messages_to_non_fncall_messages
    role, content = message['role'], message['content']
                                     ~~~~~~~^^^^^^^^^^^
KeyError: 'content'
```

**Comment by mamoodi** [comment]
I'm not familiar with this error. However, I can tell you that the model you are trying to use will never work with OpenHands.

**Comment by neubig** [comment]
Sorry about this! It should be fixed by #5158.

But as @mamoodi said, gemma-9b probably will not work with OpenHands, generally only models 70B or larger are successful at autonomous issue resolution: https://docs.all-hands.dev/modules/usage/llms
